calico_rkt/calico_rkt.py

```diff
@@ -1,138 +1,150 @@
-#!/usr/bin/env python
+# Copyright 2015 Metaswitch Networks
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
 from __future__ import print_function
 import socket
 import functools
+import logging
 import json
 import os
 import sys
 from subprocess import check_output, CalledProcessError, check_call
-
 from netaddr import IPAddress, IPNetwork, AddrFormatError
+
+from logutils import configure_logger
 from pycalico import netns
 from pycalico.ipam import IPAMClient, SequentialAssignment
 from pycalico.netns import Namespace
-from pycalico.datastore_datatypes import Rules
+from pycalico.datastore_datatypes import Rules, IPPool
 from pycalico.datastore import IF_PREFIX, DatastoreClient
 from pycalico.datastore_errors import PoolNotFound
 
-print_stderr = functools.partial(print, file=sys.stderr)
+_log = logging.getLogger(__name__)
 
-
-input_ = ''.join(sys.stdin.readlines()).replace('\n', '')
-input_json = json.loads(input_)
-
-# Append to existing env, to avoid losing PATH etc.
-# TODO-PAT: This shouldn't be hardcoded
-env = os.environ.copy()
 ETCD_AUTHORITY_ENV = 'ETCD_AUTHORITY'
-env[ETCD_AUTHORITY_ENV] = 'localhost:2379'
-
-# PROFILE_LABEL = 'CALICO_PROFILE'
-# ETCD_PROFILE_PATH = '/calico/'
 
 ORCHESTRATOR_ID = "rkt"
 HOSTNAME = socket.gethostname()
-NETNS_ROOT= '/var/lib/rkt/pods/run'
+NETNS_ROOT = '/var/lib/rkt/pods/run'
 
-def main():
+def main(env, conf_in):
     mode = env['CNI_COMMAND']
 
-    if mode == 'init':
-        print_stderr('No initialization work to perform')
-    elif mode == 'ADD':
-        print_stderr('Executing Calico pod-creation plugin')
-        create(
-            container_id=env['CNI_CONTAINERID']
-            )
+    if mode == 'ADD':
+        create(env=env, conf_in=conf_in)
     elif mode == 'DEL':
-        print_stderr('Executing Calico pod-deletion plugin')
-        delete(
-            container_id=env['CNI_CONTAINERID']
-            )
+        delete(env=env, conf_in=conf_in)
 
-def create(container_id):
+def create(env, conf_in):
     """"Handle rkt pod-create event."""
-    print_stderr('Configuring pod %s' % container_id, file=sys.stderr)
+    container_id = env['CNI_CONTAINERID']
+
+    _log.info('Configuring pod %s' % container_id)
     netns_path='%s/%s/%s' % (NETNS_ROOT, container_id, env['CNI_NETNS'])
     _datastore_client = IPAMClient()
```

Comment:
Since you only every need a single copy of `datastore_client`, and since its used at least once every time this plugin is called, it makes sense to make it a global variable. We do that a lot to the client across pycalico/calico-docker.

To remedy this, move it up to the top of this page just below your imports, keep it lower case, remove the preceeding underscore, and pull it out of your function calls.